templates/default/skel-chef-config.rb.erb

```diff
@@ -5,12 +5,11 @@ log_level                :info
 log_location             STDOUT
 cookbook_path            [".", "..", "#{current_dir}/../cookbooks"]
 
-# production
-knife[:supermarket_site] = "#{ENV['CHEF_SUPERMARKET_URL']}"
-ssl_verify_mode          :verify_none # TODO: make configurable
+knife[:supermarket_site] = "<%= @chef_supermarket %>"
+ssl_verify_mode          :<%= @ssl_verify %>
 
 # Chef Server
 node_name                "#{ENV['CHEF_SERVER_ORG_USER']}"
 client_key               "#{current_dir}/private-key.pem"
-chef_server_url          "#{ENV['CHEF_SERVER_URL']}/organizations/#{ENV['CHEF_SERVER_ORG']}"
-rubygems_url             "#{ENV['RUBYGEMS_HOST']}"
+chef_server_url          "<%= File.join(@chef_server, 'organizations') %>/#{ENV['CHEF_SERVER_ORG']}"
```

Comment:
This is a very mixed config file. Use variables from template source or use environment variables.

Reply:
Environment variables are needed for supplying the username + organization, which cannot be baked into ready-to-use CWE images. Also, when a machine is bootstrapped it cannot have manual changes in those files and env variables are a clean way to do this and cover all use cases (CWE standalone, CWE bootstrapped, plain machine)